app.js

```diff
@@ -349,6 +349,163 @@ module.exports = (app) => {
+   * @param {object} context - Probot context object
+   * @param {number} pullNumber - Pull request number
+   * @returns {Promise<Array>} - Array of approved reviews
+   */
+  async function getAllApprovedReviews(context, pullNumber) {
+    try {
+      const reviews = await context.octokit.pulls.listReviews({
+        owner: context.payload.repository.owner.login,
+        repo: context.payload.repository.name,
+        pull_number: pullNumber
+      });
+      
+      // Filter for approved reviews and get the latest review per user
+      const latestReviewsByUser = new Map();
+      
+      for (const review of reviews.data) {
+        const userId = review.user.login;
+        if (!latestReviewsByUser.has(userId) || 
+            new Date(review.submitted_at) > new Date(latestReviewsByUser.get(userId).submitted_at)) {
+          latestReviewsByUser.set(userId, review);
+        }
+      }
+      
+      // Return only approved reviews
+      return Array.from(latestReviewsByUser.values()).filter(review => review.state === 'APPROVED');
+    } catch (error) {
+      console.error('Failed to get approved reviews:', error.message);
+      return [];
+    }
+  }
+
+  /**
+   * Check if all approval criteria have been met for a pull request
+   * @param {object} context - Probot context object
+   * @param {object} pull_request - Pull request object
+   * @param {Map<string, string[]>} fileApproverMap - Map of file paths to arrays of approvers
+   * @param {Map<string, string[]>} fileTeamApproverMap - Map of file paths to arrays of team approvers
+   * @returns {Promise<{allCriteriaMet: boolean, satisfiedFiles: Set<string>, approverSummary: Array}>}
+   */
+  async function checkAllApprovalCriteriaMet(context, pull_request, fileApproverMap, fileTeamApproverMap) {
+    // Get all approved reviews
+    const approvedReviews = await getAllApprovedReviews(context, pull_request.number);
+    
+    if (approvedReviews.length === 0) {
+      return { allCriteriaMet: false, satisfiedFiles: new Set(), approverSummary: [] };
+    }
+    
+    const satisfiedFiles = new Set();
+    const approverSummary = [];
+    
+    // Check each approved reviewer
+    for (const review of approvedReviews) {
+      const satisfaction = await checkReviewerSatisfaction(
+        context, 
+        review.user.login, 
+        fileApproverMap, 
+        fileTeamApproverMap
+      );
+      
+      if (satisfaction.satisfiedFiles.length > 0) {
+        // Add satisfied files to our set
+        satisfaction.satisfiedFiles.forEach(file => satisfiedFiles.add(file));
+        
+        // Add to summary
+        approverSummary.push({
+          reviewer: review.user.login,
+          satisfiedAsIndividual: satisfaction.satisfiedAsIndividual,
+          satisfiedAsTeamMember: satisfaction.satisfiedAsTeamMember,
+          satisfiedFiles: satisfaction.satisfiedFiles
+        });
+      }
+    }
+    
+    // Check if all files are satisfied
```

Comment:
Instead of checking if all the files are satisfied by an approval, we need to check that each team and individual who were requested to review have approved and therefore satisfy our review requirements.